private/buf/bufctl/controller.go

```diff
@@ -1029,14 +1029,7 @@ func (c *controller) buildTargetImageWithConfigs(
 		if err != nil {
 			return nil, err
 		}
-		module := moduleSet.GetModuleForOpaqueID(opaqueID)
-		if module == nil {
-			return nil, syserror.Newf("new ModuleSet from WithTargetOpaqueIDs did not have opaqueID %q", opaqueID)
-		}
-		moduleReadBucket, err := bufmodule.ModuleToSelfContainedModuleReadBucketWithOnlyProtoFiles(module)
-		if err != nil {
-			return nil, err
-		}
+		moduleReadBucket := bufmodule.ModuleSetToModuleReadBucketWithOnlyProtoFiles(moduleSet)
```

Comment:
The `moduleSet` is build from the workspace targeted to the current module by `workspace.WithTargetOpaqueIDs(opaqueID)`.

Reply:
Unless I'm missing something, this is incorrect. I do not want to build the entire `ModuleSet`, I only want to build the `Module` and its `ModuleDeps`. These ware not necessarily the same thing. A `ModuleSet` may have `Modules` A, B, C, D, E, but `Module` A may only depend on B, C, and I only want to send `A, B, C` to `BuildImage`.

Reply:
This is in the comment you deleted:

```
// This function is used for cases where we want to create an Image
// specifically for one Module, such as when we need to associate LintConfig and BreakingConfig
// on a per-Module basis for buf lint and buf breaking. See bufctl.Controller, which is likely
// the only place this should be used outside of testing.
```

Reply:
If I am missing something (you hinted above with `workspace.WithTargetOpaqueIDs`), then this section needs a few paragraphs of code comments to explain why this is correct.

Reply:
We loop over each module set above `for _, module := range modules {...}` and then run `workspace.WithTargetOpaqueIDs(module.OpaqueID())` which would then take module `A` and only include it's deps `B` and `C`. So for multiple modules I believe from the comment is still kept.

Reply:
My comment is wrong, this will include more modules.

Reply:
OK, then my assumption is this needs to be reworked, correct?

Reply:
The `moduleReadBucket` may contain more modules then is needed by build image. This has changed where before the moduleSet was filtered to contain only the target and its' dependencies. However, the image must filter the `moduleReadBucket` to only target paths. So the final image will be the same as before.

The removed comment behaviour is still maintained, I've moved it to this function. An image is built for each module to associated the LintConfig and BreakingConfig on a per-module basis.